upload-api/stores/plans.js

```diff
@@ -22,7 +25,21 @@ export function usePlansStore(customerStore) {
     },
 
     set: async (customer, plan) => {
-      return await customerStore.updateProduct(customer, plan)
+      try {
+        const hasCustomerResponse = await billingProvider.hasCustomer(customer)
+        if (hasCustomerResponse.ok) {
+          await billingProvider.setPlan(customer, plan)
+          return await customerStore.updateProduct(customer, plan)
+        } else {
+          if (hasCustomerResponse.error) {
+            return hasCustomerResponse
+          } else {
+            return { error: new Failure(`billing provider does not have customer for ${customer}`) }
```

Comment:
So this would maybe be a `BillingCustomerNotFoundError` that we can use in the types for `PlansStorage`?

Reply:
ah yep - I want to replace all the `Failure`s in this PR with typed error responses like that, great catch!